ICSharpCode.CodeConverter/CSharp/NodesVisitor.cs

```diff
@@ -669,7 +669,7 @@ private IdentifierNameSyntax GetRetVariableNameOrNull(VBSyntax.MethodBlockSyntax
                 if (!AllowsImplicitReturn(node)) return null;
 
                 string methodName = node.SubOrFunctionStatement.Identifier.ValueText;
-                bool assignsToMethodNameVariable = node.Statements.Any(s => s.DescendantNodes()
+                bool assignsToMethodNameVariable = node.Statements.Any(s => s.DescendantNodesAndSelf()
```

Comment:
Probably worth a quick check to see if we can use the data flow analysis for `ReadInside` and `WrittenInside` or just `VariablesDeclared` to check for the method name in there. If not though I'll happily merge it like this.

Reply:
I've updated the code to use data flow analysis instead for this.